darwin/entry.m

```diff
@@ -155,6 +171,31 @@ void uiEntryOnChanged(uiEntry *e, void (*f)(uiEntry *, void *), void *data)
 	e->onChangedData = data;
 }
 
+// NOTE: for search widgets on OSX, setting OnFinished() alters the behaviour
+// (by setting sendsWholeSearchString).
+// Standard text and password entry widgets are not affected.
+// background:
+// On OSX, there doesn't seem to be any simple way to catch
+// both 'changed' events and 'enter' (finished editing) events on
+// search widgets. Instead, there just a single 'search' event, and flags
+// to determine when it triggers. By default, it triggers after each keypress
+// (with a little delay in case the user is still typing).
+// There's also an option to change the behaviour to trigger only when the
+// enter key is hit, or the search icon is pressed. This is the
+// sendsWholeSearchString flag, and we'll set it if (and only if) OnFinished()
+// is used.
```

Comment:
Good comment here! As for the text itself, yes, that *is* an issue, and I'm not sure what workarounds there are. The control editing system in OS X is something I don't fully understand yet, especially since NSSearchField can also use target-action. I'll have to look into it more...